### PR TITLE
[MERGE][IMP] crm{_iap_mine}{_iap_enrich}{_sms}, web, iap{_mail}: improve UI

### PR DESCRIPTION
**Pull request by amdi-odoo**
Purpose
=======
Improve some UI display of the CRM and CRM related modules.
CRM v16.3 promenade.

Specifications
===========
web
------
* Increase the maximum input field size to handle a larger number
of characters.

crm,crm_iap{_enrich}{_mine},sale,sales_team
-------------------------------------------------------------------
* Improve some small UI display like placeholders, search filters,
the wording of the lost reason modal, the fields alignments,
always underline certain fields, set certain fields as optional in
the reporting tree views, ... 

crm
------
* Modify the informative lead count field to be displayed as a muted text
sentence with text color warning when it exceeds the maximum month count.

crm{_sms}
---------------
* Remove the reporting pipeline tree view buttons as the view is not
an operational one so those won't be used by the manager.

mail
-------
* Add a m2o avatar widget to a field.

iap_mail
-------------
* Improve the display of the lead enrichment message appearing in the
chatter after an enrichment.

More details in underlying commits.

See: https://github.com/odoo/enterprise/pull/38498 

Task-3204763

### CODE REVIEW

File: addons/crm_iap_mine/static/src/scss/crm_iap_mine.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.o_crm_iap_mine_select_type .o_input {
+    padding-right: 10px;
+}
```
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/crm_iap_mine/static/src/scss/crm_iap_mine.scss:3]: Actually, it feels a bit overkill to add a file for this. I see that the widget is used in several places. Is this issue always present? This is weird if so, almost an issue by itself at widget lvl? - Separate task maybe? Can you have a look and tell my what you think ? Thx :)

File: addons/crm/report/crm_opportunity_report_views.xml [hunk 2]
```diff
@@ -1,6 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
+        <record id="crm_opportunity_report_view_tree" model="ir.ui.view">
+            <field name="name">crm.opportunity.report.tree</field>
+            <field name="model">crm.lead</field>
+            <field name="inherit_id" ref="crm.crm_case_tree_view_oppor"/>
+            <field name="mode">primary</field>
+            <field name="priority">32</field>
```
Comment (Odoonan) [inline addons/crm/report/crm_opportunity_report_views.xml:9]: Can you tell me why 32? :p

File: addons/crm/report/crm_opportunity_report_views.xml [hunk 3]
```diff
@@ -1,6 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
+        <record id="crm_opportunity_report_view_tree" model="ir.ui.view">
```
Comment (Odoonan) [inline addons/crm/report/crm_opportunity_report_views.xml:4]: CM: being picky here, it is a bit confusing. I thought you wanted to remove the buttons from the pipeline tree view, but it is only for reporting. Can you maybe reword to "remove reporting pipeline tree buttons"  - ish ?

File: addons/mail/views/mail_activity_views.xml [hunk 4]
```diff
@@ -124,7 +124,7 @@
                         </group>
                         <group>
                             <field name="date_deadline"/>
-                            <field name="user_id"/>
+                            <field name="user_id" widget="many2one_avatar_user"/>
```
Comment (Odoonan) [inline addons/mail/views/mail_activity_views.xml:127]: mismatching CM : many2many vs many2one :p

File: addons/iap_mail/static/src/scss/iap_mail.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.o_partner_autocomplete_enrich_info i {
+    min-width: fit-content;
+}
```
Comment (Odoonan) [inline addons/iap_mail/static/src/scss/iap_mail.scss:3]: missing EOF ^^

File: addons/iap_mail/data/mail_templates.xml [hunk 6]
```diff
@@ -98,12 +98,18 @@
...
-                <div t-if="twitter_bio" class="my-1 p-0 col-sm-3">
+                <div t-if="twitter_bio" class="d-flex my-1 p-0 col-sm-3">
                     <i class="fa fa-fw me-2 fa-twitter text-primary"/>
                     <b>Twitter</b>
                 </div>
                 <div t-if="twitter_bio" class="my-1 col-sm-9">
-                    <div t-if="twitter_followers"><t t-esc="twitter_followers"/> followers</div>
+                    <div class="d-flex gap-2">
+                        <a t-if="twitter" target="_blank" t-attf-href="http://www.twitter.com/{{twitter}}">
+                            http://www.twitter.com/<t t-esc="twitter"/>
```
Comment (Odoonan) [inline addons/iap_mail/data/mail_templates.xml:108]: https vs http? :question: open question :p

File: addons/iap_mail/static/src/scss/iap_mail.scss (hunk 5 above)
Comment (Odoonan) [reply addons/iap_mail/static/src/scss/iap_mail.scss:3]: While shorter, reviewers may prefer to define a class and put it on every `<i>` instead of this solution applying on the whole template. Right now, it works though, so we can keep it this way. We'll see ^^

File: addons/iap_mail/__manifest__.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
     'assets': {
         'web.assets_backend': [
             'iap_mail/static/src/js/**/*',
+            'iap_mail/static/src/scss/iap_mail.scss',
```
Comment (Odoonan) [inline addons/iap_mail/__manifest__.py:23]: CM : task number missing ^^

**Review by Odoonan — COMMENTED** [review]
Hello @amdi-odoo Good job! I've left a few comments / improvements / questions, mostly small things. Thanks for your work!

File: addons/sales_team/views/crm_team_views.xml [hunk 8]
```diff
@@ -52,7 +53,7 @@
                     <notebook>
                         <page string="Members" name="members_users">
                             <field name="member_ids" mode="kanban"
-                                class="w-100">
+                                class="w-100" context="{'allow_create_employee': False}">
```
[the remark this answers is not in the document]
Comment (amdi-odoo) [reply addons/sales_team/views/crm_team_views.xml:56]: Hello @Odoonan :slightly_smiling_face: 
Sorry, your link here is triggering me a 404 permission denied invalid sharing key :/ 
Could you send it again? :pray: 
Thanks !

[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/sales_team/views/crm_team_views.xml:56]: Sure! https://tinyurl.com/2abpbkfm shoul work :p

[the remark this answers is not in the document]
Comment (amdi-odoo) [reply addons/sales_team/views/crm_team_views.xml:56]: You're right I think it isn't related to the task as the error also happens on master :thinking: 
I got it with another field though by clicking on the domain records :
image 

Will create another task to fix it then :slightly_smiling_face: Could you fill the specifications to detail how you got your error as I couldn't reproduce yours ? :pray: Thank you !
https://www.odoo.com/web#cids=1&menu_id=4720&action=333&active_id=965&model=project.task&view_type=form&id=3267665

File: addons/crm/report/crm_opportunity_report_views.xml (hunk 2 above)
Comment (amdi-odoo) [reply addons/crm/report/crm_opportunity_report_views.xml]: Indeed not really necessary here :sweat_smile:
Removing it :slightly_smiling_face:

**Review by amdi-odoo — COMMENTED** [review]
Hello @Odoonan :slightly_smiling_face: 
Thanks for the validation! :smile: 
Just a few comments and new created tasks :slightly_smiling_face: 

Thank you!

File: addons/sales_team/views/crm_team_views.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/sales_team/views/crm_team_views.xml:56]: Sure thing! I've indicated the steps in the task :)

File: addons/crm_iap_mine/static/src/scss/crm_iap_mine.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (Odoonan) [reply addons/crm_iap_mine/static/src/scss/crm_iap_mine.scss]: Perfect, thanks for having a look :)

**Comment by Odoonan** [comment]
@amdi-odoo Hello, thanks for your updates, LGTM.
The two tasks created from here look fine too.
Greening the stage =)

**Review by awa-odoo — COMMENTED** [review]
Hello @amdi-odoo 

So it looks like I reviewed this about 2 weeks ago but never sent the comments.. oopsie.

There you go, sorry for the delay.

Cheers!

**Review by amdi-odoo — COMMENTED** [review]
Hello @awa-odoo :slightly_smiling_face: 
Updated the PR with your comments :slightly_smiling_face: 
Not sure about the line I added in upgrade though.

Thank you!

File: addons/crm/models/crm_team.py [hunk 9]
```diff
@@ -43,6 +43,9 @@ class Team(models.Model):
     lead_all_assigned_month_count = fields.Integer(
         string='# Leads/Opps assigned this month', compute='_compute_lead_all_assigned_month_count',
         help="Number of leads and opportunities assigned this last month.")
+    lead_month_assignment_exceeded = fields.Boolean('Exceed monthly lead assignement', compute="_compute_lead_all_assigned_month_count",
```
Comment (awa-odoo) [inline addons/crm/models/crm_team.py:46]: Should we do "lead_all_assigned_month_exceeded" to keep the naming aligned with the other one?

File: addons/crm/static/src/scss/crm_team.scss [hunk 10]
```diff
@@ -0,0 +1,5 @@
+.o_form_crm_team {
+    div.o_crm_lead_month_assignment_exceeded + .o_crm_lead_month_assignment {
```
Comment (awa-odoo) [inline addons/crm/static/src/scss/crm_team.scss:2]: Let's add a small comment to explain the goal, as it's a bit unusual.

File: addons/sales_team/views/crm_team_views.xml [hunk 11]
```diff
@@ -22,7 +22,7 @@
         <field name="name">crm.team.form</field>
         <field name="model">crm.team</field>
         <field name="arch" type="xml">
-            <form string="Sales Team">
+            <form string="Sales Team" class="o_form_crm_team">
```
Comment (awa-odoo) [inline addons/sales_team/views/crm_team_views.xml:25]: Let's go with "o_crm_team_form_view" actually, sorry, changed my mind :smile:

File: addons/iap_mail/static/src/scss/iap_mail.scss [hunk 12]
```diff
@@ -0,0 +1,3 @@
+.o_partner_autocomplete_enrich_info i {
+    min-width: fit-content;
```
Comment (awa-odoo) [inline addons/iap_mail/static/src/scss/iap_mail.scss:2]: Can we have a small code comment explanation on why this is necessary?

**Review by awa-odoo — COMMENTED** [review]
Hello @amdi-odoo 

Did a quick second pass, and added a few more light comments.
After that we should be good to go :rocket: 

Cheers!

**Comment by amdi-odoo** [comment]
Re @awa-odoo :slightly_smiling_face: 
What do I do about this one ? https://github.com/odoo/odoo/pull/115326#discussion_r1183462427 Do I remove the spec in the pad?

**Comment by awa-odoo** [comment]
> Re @awa-odoo slightly_smiling_face What do I do about this one ? [#115326 (comment)](https://github.com/odoo/odoo/pull/115326#discussion_r1183462427) Do I remove the spec from the pad?

You can keep the context, let's not create a bridge module just for that :slightly_smiling_face:

**Comment by awa-odoo** [comment]
@robodoo r+ rebase-merge
